src/admin/components/SearchForm.js

```diff
@@ -44,24 +44,25 @@ export class AdminSearchFormBase extends React.Component {
   }
 
   handleSearch = () => {
-    this.goToSearch(this.refs.query.value);
+    this.goToSearch(this.query.value);
   }
 
   render() {
     const { pathname, query } = this.props;
     return (
-      <form ref="form" className="search-form" method="GET"
-            action={pathname} onKeyDown={this.handleKeyDown}
+      <form method="GET" action={pathname} className="AdminSearchForm-form"
+            ref={(ref) => { this.form = ref; }} onKeyDown={this.handleKeyDown}
             onSubmit={(e) => e.preventDefault()}>
         <label className="visually-hidden" htmlFor="q">{_('Search')}</label>
-        <input ref="query" type="search" name="q" placeholder={_('Search')}
-               defaultValue={query} />
+        <input ref={(ref) => { this.query = ref; }} type="search" name="q"
```

Comment:
That said, actually, it's already more boilerplate than the string refs and that would just make it longer. The collisions would be pretty easy to spot (as they're all inside components). Maybe `query` itself is too ambiguous, but I'd rather just change this specific one to `inputQuery` but not have a generic `ref` prefix.